src/utils/argument_convert/channel.rs

```diff
@@ -0,0 +1,186 @@
+use super::ArgumentConvert;
+use crate::{model::prelude::*, prelude::*};
+
+/// Error that can be returned from [`Channel::convert`].
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum ChannelParseError {
+    /// When channel retrieval via HTTP failed
+    Http(SerenityError),
+    /// The provided channel string failed to parse, or the parsed result cannot be found in the
+    /// cache.
+    NotFoundOrMalformed,
+}
+
+impl std::error::Error for ChannelParseError {}
+
+impl std::fmt::Display for ChannelParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Http(e) => write!(f, "Failed to request channel via HTTP: {}", e),
```

Comment:
The `Display` implementation for errors should not include the message(s) of the underlying error(s). Whether to print the message of the causal error(s) should be the decision of the user. The Error Handling Work Group has a plan for improving error reports in the future via a `Report` type, but this `Display` implementation will thwart that effort, because [they do not want duplication of error information](https://blog.rust-lang.org/inside-rust/2021/07/01/What-the-error-handling-project-group-is-working-towards.html#duplicate-information-issue).
```suggestion
            Self::Http(_) => write!(f, "Failed to request channel via HTTP"),
```